### PR TITLE
Migrate Python 2 codebase to Python 3.

### PR DESCRIPTION
**Pull request by nicomiguelino**
#### Overview

- Installed [`future`](https://pypi.org/project/future/) locally
- Applied [Stage 1](https://python-future.org/automatic_conversion.html#stage-1-safe-fixes) fixes to `.py` files &ndash; `futurize --stage1 -w **/*.py` 
- Applied [Stage 2](https://python-future.org/automatic_conversion.html#stage-2-py3-style-code-with-wrappers-for-py2) fixes to .py files – `futurize --stage2 -w **/*.py`
- Used `python3` in replacement of `python`
- Pinned versions of some dependencies (`requirements.txt`, for instance)
- Fixed unicode/bytes issues when migrating code from Python 2 to Python 3.

#### Custom Installation Command (Testing)

```bash
cd && \
echo "export CUSTOM_BRANCH='py-2-to-3'" > .env && \
bash <(curl -sL https://raw.githubusercontent.com/nicomiguelino/Anthias/custom-install/bin/install.sh)
```

### CODE REVIEW

File: bin/build_containers.sh [hunk 1]
```diff
@@ -59,16 +59,20 @@ for container in server celery redis websocket nginx viewer wifi-connect 'test';
         export CHROMEDRIVER_DL_URL="https://chromedriver.storage.googleapis.com/107.0.5304.62/chromedriver_linux64.zip"
 
     elif [ "$container" == 'wifi-connect' ]; then
+        if [ "$BOARD" == 'x86' ]; then
+            continue
+        fi
```
Comment (nicomiguelino) [inline bin/build_containers.sh:64]: I can remove this block later and make the if-statement below support x86.

**Comment by nicomiguelino** [comment]
Here's a recent viewer log...

```
screenly-srly-ose-viewer-1  | ./bin/start_viewer.sh: line 39: /sys/fs/cgroup/memory/memory.swappiness: No such file or directory
screenly-srly-ose-viewer-1  | Loading browser...
screenly-srly-ose-viewer-1  | Viewer crashed.
screenly-srly-ose-viewer-1  | Traceback (most recent call last):
screenly-srly-ose-viewer-1  |   File "viewer.py", line 532, in <module>
screenly-srly-ose-viewer-1  |     main()
screenly-srly-ose-viewer-1  |   File "viewer.py", line 501, in main
screenly-srly-ose-viewer-1  |     setup()
screenly-srly-ose-viewer-1  |   File "viewer.py", line 428, in setup
screenly-srly-ose-viewer-1  |     load_browser()
screenly-srly-ose-viewer-1  |   File "viewer.py", line 324, in load_browser
screenly-srly-ose-viewer-1  |     while 'Screenly service start' not in browser.process.stdout:
screenly-srly-ose-viewer-1  | TypeError: a bytes-like object is required, not 'str'
screenly-srly-ose-viewer-1  | Traceback (most recent call last):
screenly-srly-ose-viewer-1  |   File "viewer.py", line 532, in <module>
screenly-srly-ose-viewer-1  |     main()
screenly-srly-ose-viewer-1  |   File "viewer.py", line 501, in main
screenly-srly-ose-viewer-1  |     setup()
screenly-srly-ose-viewer-1  |   File "viewer.py", line 428, in setup
screenly-srly-ose-viewer-1  |     load_browser()
screenly-srly-ose-viewer-1  |   File "viewer.py", line 324, in load_browser
screenly-srly-ose-viewer-1  |     while 'Screenly service start' not in browser.process.stdout:
screenly-srly-ose-viewer-1  | TypeError: a bytes-like object is required, not 'str'
```

**Comment by nicomiguelino** [comment]
> Here's a recent viewer log...
> 
> ```
> screenly-srly-ose-viewer-1  | ./bin/start_viewer.sh: line 39: /sys/fs/cgroup/memory/memory.swappiness: No such file or directory
> screenly-srly-ose-viewer-1  | Loading browser...
> screenly-srly-ose-viewer-1  | Viewer crashed.
> screenly-srly-ose-viewer-1  | Traceback (most recent call last):
> screenly-srly-ose-viewer-1  |   File "viewer.py", line 532, in <module>
> screenly-srly-ose-viewer-1  |     main()
> screenly-srly-ose-viewer-1  |   File "viewer.py", line 501, in main
> screenly-srly-ose-viewer-1  |     setup()
> screenly-srly-ose-viewer-1  |   File "viewer.py", line 428, in setup
> screenly-srly-ose-viewer-1  |     load_browser()
> screenly-srly-ose-viewer-1  |   File "viewer.py", line 324, in load_browser
> screenly-srly-ose-viewer-1  |     while 'Screenly service start' not in browser.process.stdout:
> screenly-srly-ose-viewer-1  | TypeError: a bytes-like object is required, not 'str'
> screenly-srly-ose-viewer-1  | Traceback (most recent call last):
> screenly-srly-ose-viewer-1  |   File "viewer.py", line 532, in <module>
> screenly-srly-ose-viewer-1  |     main()
> screenly-srly-ose-viewer-1  |   File "viewer.py", line 501, in main
> screenly-srly-ose-viewer-1  |     setup()
> screenly-srly-ose-viewer-1  |   File "viewer.py", line 428, in setup
> screenly-srly-ose-viewer-1  |     load_browser()
> screenly-srly-ose-viewer-1  |   File "viewer.py", line 324, in load_browser
> screenly-srly-ose-viewer-1  |     while 'Screenly service start' not in browser.process.stdout:
> screenly-srly-ose-viewer-1  | TypeError: a bytes-like object is required, not 'str'
> ```

The issue was solved by converting browser.process.stdout from `bytes` to `str`.

File: requirements/requirements.viewer.txt [hunk 2]
```diff
@@ -1,10 +1,13 @@
 Cython==0.29.33
 Flask==1.1.2
+Jinja2==2.11.3
```
Comment (nicomiguelino) [inline requirements/requirements.viewer.txt:3]: Sort the the requirements file alphabetically. Apply to all other requirements file.

File: bin/migrate.py [hunk 3]
```diff
@@ -78,9 +80,9 @@ def open_db_get_cursor():
 def migrate_add_column(col, script):
     with open_db_get_cursor() as (cursor, conn):
         if test_column(col, cursor):
-            print 'Column (' + col + ') already present'
+            print('Column (' + col + ') already present')
```
Comment (nicomiguelino) [inline bin/migrate.py:83]: Make use of [f-strings](https://docs.python.org/3/whatsnew/3.6.html#whatsnew36-pep498) (which was introduced in Python 3.6) instead. It's much more cleaner.

File: docker-compose.yml.tmpl [hunk 4]
```diff
@@ -24,9 +19,6 @@ services:
 
   anthias-server:
     image: screenly/anthias-server:${DOCKER_TAG}-${DEVICE_TYPE}
-    build:
-      context: .
-      dockerfile: docker/Dockerfile.server
```
Comment (nicomiguelino) [inline docker-compose.yml.tmpl:29]: I'm not sure if I will remove this one. Running `docker compose` with the compose file generated from this template will fail. At the moment, I'm just running `bin/build_containers.sh` when re-building the images/services.

File: docker-compose.yml.tmpl [hunk 5]
```diff
@@ -128,16 +111,19 @@ services:
...
       - HOME=/data
     depends_on:
       - anthias-server
       - anthias-websocket
+    networks:
+      default:
+      anthias-extra-network-1:
+        aliases:
+          - 'anthias-nginx'
+          - 'srly-ose-nginx'
```
Comment (nicomiguelino) [inline docker-compose.yml.tmpl:126]: Maybe add a comment to remove the changes once a new version of the WebView has been released.

File: docker-compose.yml.tmpl [hunk 6]
```diff
@@ -5,8 +5,6 @@ services:
   anthias-wifi-connect:
     image: screenly/anthias-wifi-connect:${DOCKER_TAG}-${DEVICE_TYPE}
     build:
-      args:
```
Comment (vpetersson) [inline docker-compose.yml.tmpl:8]: Why was this removed?

**Review by vpetersson — APPROVED** [review]
Approved with a minor change.
